src/pages/list/[pageId].tsx

```diff
@@ -54,6 +59,25 @@ const Page: NextPage<PageProps> = ({ data }) => {
     }
   };
 
+  const filterBySearch = () => {
+    let updatedList = [...data];
```

Comment:
This is great. However there is one thing that's not optimal. You are filtering the page data with this. If there are 50 Pokemon on the page, then you filter just within those 50 and not within the 1000+ pokemon that exist.
This is a bit more complicated to solve as you can not put all 1000+ Pokemon in the page data. We can do a session together to think about possible solutions. :)